### ISSUE TITLE
make gl::ScopedAlphaBlend more versatile

### ISSUE THREAD

**Issue by richardeakin** [cinder/Cinder#629]
I've been using this frequently as of late, and I find the following:

```
gl::ScopedAlphaBlend blendScope( false );
```

to be conterintuitive, as when I read that I see it as 'disabling alpha blend', when in fact it means premultipled alpha.

What about having a `gl::ScopedBlend` object that takes an enum as value, instead of bool? Something like:

```
enum class BlendMode { ALPHA, ALPHA_PREMUL, ADDITIVE };

gl::ScopedBlend    blendScope( BlendMode::ALPHA );
```

The `ScopedAdditiveBlend` class would be folded into this..

**Comment by gaborpapp** [comment]
This looks nice. There might be a use case where you want to disable blending in the scope. In this case another enum value is needed, for example ```BlendMode::NONE```.

**Comment by paulhoux** [comment]
I like this approach, especially if we could write the following:
```gl::ScopedBlend blend( NONE );```

To avoid confusion, I would suggest to remove ```ScopedAdditiveBlend``` and ```ScopedAlphaBlend```.

**Comment by richardeakin** [comment]
After discussing this here, we've resolved to hold off introducing a new enum into the ci::gl namespace, though perhaps a `BlendMode` enum will find its way into future shader work. There actually already is a [ScopedBlend](https://github.com/cinder/Cinder/blob/glNext/include/cinder/gl/scoped.h#L78), I suppose I didn't find it because I was thinking 'alpha' in my head and hit the `ScopedAlphaBlend` version.

To improve on this, we're going to rename the other RAII blending structs so they all follow the form `ScopedBlend*`, and add a new one `ScopedBlendPremult` to remove the confusion within `ScopedAlphaBlend`.

**Comment by richardeakin** [comment]
Fixed with #876
